biz.aQute.bndlib/src/aQute/bnd/service/reporter/ReportImportDeserializerPlugin.java

```diff
@@ -0,0 +1,31 @@
+package aQute.bnd.service.reporter;
+
+import java.io.InputStream;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * This plugin deserialize a stream of a specific format into a DTO
+ * representation.
+ */
+@ProviderType
+public interface ReportImportDeserializerPlugin {
```

Comment:
Isn't this more generally `ReportDeserializerPlugin`?

Reply:
We do not deserialize a report, but user can import file content to their report (eg, pom.xml, or pom.properties). So this plugin is made to deserialize imported file.

Reply:
I guess I don't see the larger design into which all these plugins fit. Perhaps some high level documentation in package-info is in order to explain how all these plugins interact and are expected to be used. The lack of this may be the cause of many of my misunderstandings.

Reply:
I think the global use is described in the PR #2169, but for the deserializer plugin, here is a specific example:

- Suppose you want to document all your projects that produce an executable application jar, your applications use the OSGI Configuator  service (with a `OSGI-INF/configurator/configuration.json` ressource) and you want to document the provided configuration data into a readme file.
- A Json ReportImportDeserializerPlugin implementation will be used to import the `OSGI-INF/configurator/configuration.json` ressource into the report DTO.
- The user is then free to use XSLT or twig (or any other) template files to generate its readme file. If he uses XSLT, the report DTO will be serialized to XML with the ReportSerializerPlugin and the result will be provided to a XSLT implementation of the ReportTransformerPlugin. The user can also just serialize the report into a file and use this file with an external tool.

Reply:
I may be described in that issue, but the issue is not part of the code base (and this PR). I think we need that sort of overall doc in the package's package-info.java file.

Reply:
So the name of the plugin confused me. It is not deserializing a report. It is converting some file into a DTO.

In reading the text in #2169, is says

    aQute.bnd.service.reporter.ReportImportDeserializerPlugin: Convert a file into a DTO representation. Just like ProjectEntryPlugin converts a project into a DTO. There is no report here, no import and no deserialization here. 

Perhaps a better name for this type is ResourceEntryPlugin whose method is

`public Object extract(InputStream input) throws Exception;`

Reply:
Yes it is a better name.